sdk/core/src/error/http_error.rs

```diff
@@ -26,7 +26,7 @@ impl HttpError {
         let body = response.into_body().await;
         error_code = error_code.or_else(|| get_error_code_from_body(&body));
         HttpError {
-            status: status.as_u16(),
+            status: status as u16,
```

Comment:
Since we are re-exporting StatusCode now, can we make the error include the enum value instead of a u16?  This will make dealing with contextualized errors significantly more straightforward.

Reply:
added #870